account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py

```diff
@@ -129,15 +129,15 @@ def _compute_unrealized_currency_gl(self, currency_id, balances, form):
         ctx_rate['date'] = form.revaluation_date
         cp_currency = form.journal_id.company_id.currency_id
 
-        currency = currency_obj.browse(currency_id)
+        currency = currency_obj.browse(currency_id).with_context(ctx_rate)
 
         foreign_balance = adjusted_balance = balances.get(
             'foreign_balance', 0.0)
         balance = balances.get('balance', 0.0)
         unrealized_gain_loss = 0.0
         if foreign_balance:
             ctx_rate['revaluation'] = True
```

Comment:
By moving the assignation of the context up this line would have no effect, no ?

Reply:
In such case the context is defined on `currency` var, so that it's not only used on the `compute` call but also when we get the rate here : https://github.com/OCA/account-closing/pull/101/files/3aaae75184fb6c3487270c4f23de8a1008eda73d#diff-2f2adec7633819d0a28eb35fef7fca93R150

Reply:
Yes, but does changing the variable used for context after it has been assigned actually impacts the context ?
I don't think so.

Reply:
Sorry for the confusion but you are right.
However, as I looked into this context's usage, it was actually not needed, and I added the following commit to remove it.

Reply:
Simplified code then :smile: